### PR TITLE
[bug-1205]: Fix Authorization Server install in non-authorization namespace

### PR DESCRIPTION
**Pull request by atye**
# Description

- Fixes https://github.com/dell/csm/issues/1205
- Removes Authorization v1.8.0 and adds v1.11.0
- Moves karavi-storage-secret.yaml to deployment.yaml so user does not have to create it
- Removes the namespace field from the Authorization sample file to support custom namespace
- Makes namespace configurable in Authorization kube-mgmt policy configuration and relevant cluster role and bindings

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1205 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

There are no go code changes here so unit tests are not relevant. Since the bug is a runtime bug when creating volumes, e2e doesn't cover this yet. 

This was manually tested by installing Authorization in a non-authorization namespace, installing csi-powerflex, and creating volumes.

### CODE REVIEW

File: samples/authorization/karavi-storage-secret.yaml [hunk 1]
```diff
@@ -1,8 +0,0 @@
-apiVersion: v1
```
Comment (shaynafinocchiaro) [inline samples/authorization/karavi-storage-secret.yaml:1]: Any changes required in the E2E tests if this secret manifest is removed?

File: operatorconfig/moduleconfig/authorization/v1.11.0/container.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: karavi-authorization-proxy
 imagePullPolicy: IfNotPresent
-image: dellemc/csm-authorization-sidecar:v1.8.0
+image: dellemc/csm-authorization-sidecar:v1.11.0
```
Comment (alikdell) [inline operatorconfig/moduleconfig/authorization/v1.11.0/container.yaml:3]: isn't last release is 1.10.0?

File: samples/authorization/csm_authorization_proxy_server_v1110.yaml [hunk 3]
```diff
@@ -2,32 +2,31 @@ apiVersion: storage.dell.com/v1
...
   name: authorization
-  namespace: authorization
 spec:
   modules:
     # Authorization: enable csm-authorization proxy server for RBAC
     - name: authorization-proxy-server
       # enable: Enable/Disable csm-authorization
       enabled: true
-      configVersion: v1.8.0
+      configVersion: v1.11.0
```
Comment (alikdell) [inline samples/authorization/csm_authorization_proxy_server_v1110.yaml:11]: 1.10.0?

File: operatorconfig/moduleconfig/authorization/v1.11.0/container.yaml (hunk 2 above)
Comment (jooseppi-luna) [reply operatorconfig/moduleconfig/authorization/v1.11.0/container.yaml:3]: backtothefuture

**Review by jooseppi-luna — COMMENTED** [review]
controller unit tests are failing 🤔 and I have the same question as Shayna about e2e

**Pull request closed without merging**